Document Keyboard Shortcuts

hannydevelop opened the pull request:
This is a draft pull request, Kindly look at the changes I have made and places you think I should improve on. I am so glad to be working on this!

Resolves #38.

Be-ing commented:
add to whitelist

Be-ing commented on source/chapters/appendix.rst:
```diff
@@ -19,6 +19,12 @@ There are a variety of other resources where you can find out more:
   `<https://mixxx.org/wiki/>`_
 
+* **Keyboard Shortcuts**: This a list for all default Mixxx keyboard shortcuts. 
+  Please note that this shortcuts are for keyboards configured in English
```
```suggestion
  Please note that these shortcuts are for keyboards configured in English
```

Be-ing commented on source/chapters/appendix.rst:
```diff
@@ -19,6 +19,12 @@ There are a variety of other resources where you can find out more:
 
+* **Keyboard Shortcuts**: This a list for all default Mixxx keyboard shortcuts. 
+  Please note that this shortcuts are for keyboards configured in English
+  layout and default shortcuts may be different if your language settings
```
```suggestion
  layout and default shortcuts may be different if your language setting
```

Be-ing commented on source/chapters/effects.rst:
```diff
@@ -27,6 +27,9 @@ processed one after the other in series. You will hear different sounds
 depending on the order of effects within the chain. Each effect in the chain
 has its own button to toggle it on and off.
 
+The shortcut to quickly Activate/Disable Effects Unit is 5 Deck1 and 0 in Deck2 
+in an English keyboard layout.
```
Which effect unit is activated by each of these shortcuts?

Also, I think this would be better to put at the bottom of the section after the signal flow through the effect units is explained.

ronso0 replied:
Those are the QuickEffect shortcuts, so wrong context.
addressed by later review.

Be-ing commented on source/chapters/appendix.rst:
```diff
@@ -19,6 +19,12 @@ There are a variety of other resources where you can find out more:
+* **Keyboard Shortcuts**: This a list for all default Mixxx keyboard shortcuts. 
+  Please note that this shortcuts are for keyboards configured in English
+  layout and default shortcuts may be different if your language settings
+  is not english.
```
```suggestion
  is not English.
```

Be-ing commented on source/chapters/introduction.rst:
```diff
@@ -27,6 +27,9 @@ is appropriate for the present moment and that will mix well into the
 separate audio outputs, typically provided by a USB audio interface (also known
 as a “sound card”, although most of them are not shaped like cards anymore).
+The shortcut for headphone cueing in Deck1 is T while for Deck 2 the shortcut is Y. 
+The short cut is for English keyboard layout, settings may be different depending on
+your language settings.
```
Repeating this everywhere a hotkey is documented is tedious. I'm sure there's a better way to do this. Can we make `shortcut` a glossary term and explain in the glossary that the shortcuts in the manual are only for English keyboards?

hannydevelop replied:
I think this would be nice, I am on it now!

ronso0 replied:
yeah, printing this hint once at the top of the Keyboard chapter should suffice.
```suggestion
The shortcut for headphone cueing in Deck1 is :kbd:`T` while for Deck 2 the shortcut is :kbd:`Y`. 
```

ronso0 replied:
As stated in the current review, let's _not_ add an appendix paragraph but link to Controlling Mixxx > keyboard instead and mention the English lyout there.

Be-ing commented on source/chapters/library.rst:
```diff
@@ -114,7 +114,7 @@ Tracks - View and edit your whole collection
 **Loading tracks**
   To load a track into a :ref:`deck <interface-decks>`, you can either drag it
-  to the waveform display or use the context menu. Go to the chapter
+  to the waveform display, use the context menu or use shortcut keys. Go to the chapter
```
```suggestion
  to the waveform display, use the context menu, or use shortcut keys. Go to the chapter
```

hannydevelop commented:
Thank you so much for the reviews @Be-ing I would effect the changes and create a pull request as soon as possible. Thank for the speedy response and guidance.

Be-ing commented on source/chapters/library.rst:
```diff
@@ -324,6 +326,14 @@ Tracks can be loaded into a deck in several ways:
   Mixxx's waveform displays will load it into that deck.
 
+
+unLoading Tracks
```
```suggestion
Unloading Tracks
```

Be-ing commented on source/chapters/library.rst:
```diff
@@ -324,6 +326,14 @@ Tracks can be loaded into a deck in several ways:
+
+unLoading Tracks
+==============
+
```
If we're adding a section for this, the eject buttons in the decks should also be mentioned.

hannydevelop replied:
Thank you for the guide!

Be-ing commented:
Please don't make a new pull request. Make new commits on this branch then push them to GitHub and the new commits will be added to this pull request. Also, please [write more descriptive commit messages](https://chris.beams.io/posts/git-commit/).

hannydevelop commented:
> Please don't make a new pull request. Make new commits on this branch then push them to GitHub and the new commits will be added to this pull request. Also, please [write more descriptive commit messages](https://chris.beams.io/posts/git-commit/).

I would do as instructed! Thank you!

hannydevelop commented:
Hello @Be-ing I just implemented the changes! If there is something you would love to draw my attention to, I would be more than glad to look at it.

hannydevelop commented on source/chapters/library.rst:
```diff
@@ -327,12 +327,17 @@ Tracks can be loaded into a deck in several ways:
+* To unload track in Deck 1, use the keyboard shortcut Shift + Left. In Deck 2 you can
   unload track with Shift + Right
 
+* You can use the eject track button to unload track from a deck. This button is
+  located at the bottom right, below the waveform.
```
I would implement this now @ronso0  . Thank you so much for taking out time to review this.

hannydevelop replied:
Hello @ronso0 I just implemented this, but i don't really know how to implement this `QT_SCREEN_SCALE_FACTORS=2` today is the first time I am using RST actively. The first time I used it involved no addition of pictures. What do you think about the coloured eject button? since you wanted more emphasis on it

ronso0 replied:
oh sorry, forgot you're ne wto mixxx :)
scale factors refers to running Mixxx with an user-specified scaled GUI, for example on 4k screens.
Instructions are here:
https://github.com/mixxxdj/mixxx/wiki/troubleshooting#graphical-user-interface-gui-is-too-big-or-too-small
If you don't succeed, let me know. I can upload the picture myself later today.
> What do you think about the coloured eject button?

either blur the sorroundungs or desaturate everything except the button and use a colored icon.

hannydevelop replied:
> oh sorry, forgot you're ne wto mixxx :)
> scale factors refers to running Mixxx with an user-specified scaled GUI, for example on 4k screens.
> Instructions are here:
> https://github.com/mixxxdj/mixxx/wiki/troubleshooting#graphical-user-interface-gui-is-too-big-or-too-small
> If you don't succeed, let me know. I can upload the picture myself later today.
> 
> > What do you think about the coloured eject button?
> 
> either blur the surroundings or desaturate everything except the button and use a colored icon.

Hello @ronso0 I read the [instructions](https://github.com/mixxxdj/mixxx/wiki/troubleshooting#graphical-user-interface-gui-is-too-big-or-too-small) and can't seem to understand how to implement the scale factor. The instruction explains why an image maybe too big or small and how to remove scale factor from your device. can you please do it so I can learn from what you did? Thank you!

ronso0 replied:
This is what I had in mind.
![Mixxx-eject-button png](https://user-images.githubusercontent.com/5934199/95598055-bdec3480-0a4f-11eb-9b06-2dcc4762bf8a.png)
or this
![Mixxx-eject-button png_blur](https://user-images.githubusercontent.com/5934199/95598573-5f738600-0a50-11eb-90e6-63bf23dbfcab.png)

We may not need the entire deck, so feel free to crop or tweak the image further, as you like. Check what looks best in the manual context.
Then you may `git add [path to image] && git commit --amend` so it's part of your previous commit "Adding image of eject button".
`git push -f` afterwards.

hannydevelop replied:
Hello @ronso0 the first Image is so nice! I would work on this now. Hoping to learn more from you! I really like the first image.

hannydevelop replied:
Hello @ronso0 I just implemented this now! What do you think?

hannydevelop replied:
@ronso0 I ran ```git commit -m```  before ```git --amend -m```. It was a mistake, am sorry

Holzhaus replied:
Note that this PR goes into the `manual-2.2.x` branch, so the Screenshots should match the default skin used in Mixxx 2.2. We'll need to update the Screenshots later for Mixxx 2.3.

ronso0 replied:
okay, so let's leave the screenshot out of this
@hannydevelop unless you're willing to install Mixxx 2.2.4 and prepare a screenshot with the Deere skin similiar to what I posted above.
but we can also add that after #154 has been merged.
As you like..

hannydevelop replied:
> okay, so let's leave the screenshot out of this
> @hannydevelop unless you're willing to install Mixxx 2.2.4 and prepare a screenshot with the Deere skin similiar to what I posted above.
> but we can also add that after #154 has been merged.
> As you like..

Ok, I would remove the Image then

hannydevelop replied:
> > okay, so let's leave the screenshot out of this
> > @hannydevelop unless you're willing to install Mixxx 2.2.4 and prepare a screenshot with the Deere skin similiar to what I posted above.
> > but we can also add that after #154 has been merged.
> > As you like..
> 
> Ok, I would remove the Image then

Image has been removed

ronso0 replied:
please also remove the lonely `Unloading Tracks` header.

ronso0 replied:
Updated by current review

ronso0 replied:
Eject button image can be added later on, keyboard should be mentioned

hannydevelop commented:
Hello @ronso0 and @Holzhaus can final review be made and this Pull Request merged? I need to open a pull request for https://github.com/mixxxdj/manual/issues/137

Holzhaus commented:
@hannydevelop we'll have a look soon, but you can work in a separate branch and open a PR for that one. That way you don't have to wait for this PR to be merged.

https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell

ronso0 commented on source/chapters/effects.rst:
```diff
@@ -61,6 +61,9 @@ in the unit, there will be comb filtering which is probably not desired. For
 more background information, read `Equalizers and Phase Shift
 <http://ethanwiner.com/EQPhase.html>`_.
 
+The shortcut to quickly activate/disable any effect unit for Deck 1 is 5 and 0 for Deck 2 
+in an English keyboard layout.
```
Those shortcuts control the single Quick Effects (green) which are located in the mixer
![image](https://user-images.githubusercontent.com/5934199/95658740-edb53e00-0b1c-11eb-96ae-0d2ce10028e3.png)
I suggest you remove this change. Unfortunately there's no mention of the Quick Effects, yet (just created https://github.com/mixxxdj/manual/issues/141 to cover that)

hannydevelop replied:
> Those shortcuts control the single Quick Effects (green) which are located in the mixer
> ![image](https://user-images.githubusercontent.com/5934199/95658740-edb53e00-0b1c-11eb-96ae-0d2ce10028e3.png)
> I suggest you remove this change. Unfortunately there's no mention of the Quick Effects, yet (just created #141 to cover that)

Change has been removed

hannydevelop replied:
> Those shortcuts control the single Quick Effects (green) which are located in the mixer
> ![image](https://user-images.githubusercontent.com/5934199/95658740-edb53e00-0b1c-11eb-96ae-0d2ce10028e3.png)
> I suggest you remove this change. Unfortunately there's no mention of the Quick Effects, yet (just created #141 to cover that)

done!

ronso0 commented:
Keyboard shortcuts are highlighted in the text. Currently we use `:kbd:` but I think that doesn't stand out appropriately and we should improve that (not in this PR though :)
```
then press :kbd:`Ctrl` + :kbd:`T`
```

hannydevelop commented:
> Keyboard shortcuts are highlighted in the text. Currently we use `:kbd:` but I think that doesn't stand out appropriately and we should improve that (not in this PR though :)
> 
> ```
> then press :kbd:`Ctrl` + :kbd:`T`
> ```

Can you open an issue concerning this and assign that to me while we merge/close this one?

ronso0 commented:
> Can you open an issue concerning this and assign that to me while we merge/close this one?

Here yo go https://github.com/mixxxdj/manual/issues/145
Thanks!

hannydevelop commented:
> > Can you open an issue concerning this and assign that to me while we merge/close this one?
> 
> Here yo go #145
> Thanks!
> @Holzhaus Looks like I can't assign issues for soem reason..

Ok! I will comment on the issue now! all changes requested for have been effected. Any new changes? I will be glad to work on them till its perfect

hannydevelop commented:
> @hannydevelop we'll have a look soon, but you can work in a separate branch and open a PR for that one. That way you don't have to wait for this PR to be merged.
> 
> https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell

I just did, thanks for the information! it was a great help

hannydevelop commented:
Hello @Holzhaus are you still reviewing this PR or are there changes you suggest i make on this PR?

Holzhaus reviewed (changes requested):
Looks like your last commit accidently added the unrelated "Import from Serato" chapter. Could you amend that commit and remove the unrelated changes please?

hannydevelop commented:
> Looks like your last commit accidently added the unrelated "Import from Serato" chapter. Could you amend that commit and remove the unrelated changes please?

ok! I would do that now!

hannydevelop commented:
> Looks like your last commit accidently added the unrelated "Import from Serato" chapter. Could you amend that commit and remove the unrelated changes please?

Hello @Holzhaus I just did this, please let me know if there are any changes you would want me to implement. I would be glad to work on them.

Holzhaus commented on source/index.rst:
```diff
@@ -47,6 +47,7 @@ Table of Contents
    chapters/effects
    chapters/djing_with_mixxx
    chapters/preferences
+   chapters/import_from_serato
```
Please remove this and amend the commit.

hannydevelop replied:
Hello @Holzhaus can you give me sometime to fix this? I used ```git reflog``` and ```git reset head``` inappropriately and my working tree is messed up

Holzhaus replied:
You can reset your working tree to the version you pushed here, by using

    git reset --hard origin/manual-2.2.x

(You should probably make a backup first by using `git branch -c mybackup`).

Next, you need to find out which commit introduced the unwanted change. By using `git log` and `git show`, you can easily check that it was this one: https://github.com/mixxxdj/manual/commit/41e0e66f10f736aa1df691e66f2e0e3c7c36c7b2#diff-c05e5dd7e57e7292e7fccc39bebe079e9a2062e0c9a3a57a4b6c608bd8d5669b

Since it's the last commit, you can just go ahead and remove that line, than run `git commit --amend`.

hannydevelop replied:
Thank you so much for the help @Holzhaus I ran into some problems too while doing this ```git reset --hard origin/manual-2.2.x```. I resolved into updating the files manually since the initial reset didn't wipe everything away. please review. Thank you!

Holzhaus commented:
I think you just reverted a bunch of other stuff accidently, you can just run `git rebase -i HEAD~2` and replace `pick` with `drop` in front of the last commit to undo it.

hannydevelop commented:
> I think you just reverted a bunch of other stuff accidently, you can just run `git rebase -i HEAD~2` and replace `pick` with `drop` in front of the last commit to undo it.

yes, exactly what I did , i would run this now!

hannydevelop commented:
> > I think you just reverted a bunch of other stuff accidently, you can just run `git rebase -i HEAD~2` and replace `pick` with `drop` in front of the last commit to undo it.
> 
> yes, exactly what I did , i would run this now!

Hello @Holzhaus I have corrected everything and implemented the changes, please review it!

hannydevelop commented:
> Looks like your last commit accidently added the unrelated "Import from Serato" chapter. Could you amend that commit and remove the unrelated changes please?

done!

ronso0 commented on source/chapters/library.rst:
```diff
@@ -324,6 +326,24 @@ Tracks can be loaded into a deck in several ways:
   Mixxx's waveform displays will load it into that deck.
 
+
+UnLoading Tracks
```
```suggestion
Unloading Tracks
```

ronso0 commented on source/glossary.rst:
```diff
@@ -8,6 +8,12 @@ Glossary of Terms
      cueing in your headphones. Being able to cue is a crucial aspect to DJing.
+     The shortcut for headphone cueing in Deck1 is T while for Deck 2 the shortcut is Y.
+
+   shortcuts
```
referencing @Be-ing 's hint: please link this term in the manual (where it make sense) with :term:`shortcuts <shortcuts>` or :term:`keyboardhortcuts <shortcuts>` (depending on the context)

hannydevelop replied:
> referencing @Be-ing 's hint: please link this term in the manual (where it make sense) with :term:`shortcuts <shortcuts>` or :term:`keyboardhortcuts <shortcuts>` (depending on the context)

Can you please explain what you meant here? I don't really understand. Thank you!

ronso0 replied:
We use :term:"display name <term_in_gloassary>" to link terms to respective paragraph in the Glossary
For example 'controllers' in https://github.com/mixxxdj/manual/blob/manual-2.2.x/source/chapters/controlling_mixxx.rst#using-a-keyboard links to https://www.mixxx.org/manual/latest/en/glossary.html#term-controller
(in the above exapmle replace **"** with **`** which unfortunately is the markdown shortcut for 'code' in github comments)

Holzhaus replied:
@ronso0 just put the code on a new line and indent by four spaces, then you can use

    `

inside code blocks.

Holzhaus replied:
```suggestion
   shortcut
   keyboard shortcut
```

Let's use the singular form here.

ronso0 commented:
@hannydevelop Please resolve conversations only after having commited and pushed the requested changes.
If there's a disagreement about changes, the requestor should resolve it if desired.

Btw, you can adopt code suggestions from reviewers directly either with `Coomit suggestion` or collecting several suggestions to merge them in a single commit:
![image](https://user-images.githubusercontent.com/5934199/96018413-84e70200-0e4b-11eb-931a-23bd8e006c4d.png)

ronso0 commented:
Please use the rst selector :kbd: to highlight keyboard shortcuts, for example :kbd:`Enter`

https://github.com/mixxxdj/manual/blame/manual-2.2.x/source/chapters/user_interface.rst#L642

hannydevelop commented:
> @hannydevelop Please resolve conversations only after having commited and pushed the requested changes.
> If there's a disagreement about changes, the requestor should resolve it if desired.
> 
> Btw, you can adopt code suggestions from reviewers directly either with `Coomit suggestion` or collecting several suggestions to merge them in a single commit:
> ![image](https://user-images.githubusercontent.com/5934199/96018413-84e70200-0e4b-11eb-931a-23bd8e006c4d.png)

Sure, that's what I did. The issue here is my working tree got messed up because I used the ```reflog``` and ```reflag``` command inappropriately and literally had to start from the beginning since most of the changes i made were overwritten.

ronso0 commented:
oh, sounds...not fun :\

hannydevelop commented:
> oh, sounds...not fun :\

Certainly not fun. I just implemented the reviews you made, please look at it.

ywwg commented:
> Certainly not fun. I just implemented the reviews you made, please look at it.

this happens to everyone in the beginning.  It's unfortunate but thanks for redoing the work so quickly

hannydevelop commented:
> > Certainly not fun. I just implemented the reviews you made, please look at it.
> 
> this happens to everyone in the beginning. It's unfortunate but thanks for redoing the work so quickly

Thank you! I appreciate the kind words, i needed them now

ronso0 commented:
Unfortunately the preview is updated, yet, which makes this a bit hard to review.
@hannydevelop did yo maybe not push your recent changes, yet?

hannydevelop commented:
> Unfortunately the preview is updated, yet, which makes this a bit hard to review.
> @hannydevelop did yo maybe not push your recent changes, yet?

I just did. mehn, i committed and forgot to push

ronso0 commented:
alright then, I'll check that tomorrow. Thanks!

hannydevelop commented:
> alright then, I'll check that tomorrow. Thanks!

Thank you too!

deborahtrez commented:
Hey @hannydevelop , the deploy preview for this PR is failing to show.

![image](https://user-images.githubusercontent.com/28389950/96109977-eceb2600-0ee7-11eb-82c1-201bdebf93f2.png)
 

I had the same issue just yesterday so @Holzhaus asked me to pull the latest changes  from the main repo with `git pull https://github.com/mixxxdj/manual.git manual-2.2.x`  so that review will be simpler. Pull then push to your forked repo. That should do the trick :)

hannydevelop commented:
> Hey @hannydevelop , the deploy preview for this PR is failing to show.
> 
> ![image](https://user-images.githubusercontent.com/28389950/96109977-eceb2600-0ee7-11eb-82c1-201bdebf93f2.png)
> 
> I had the same issue just yesterday so @Holzhaus asked me to pull the latest changes from the main repo with `git pull https://github.com/mixxxdj/manual.git manual-2.2.x` so that review will be much simpler. Pull from the main repo then push to your forked repo. That should do the trick :)

Sure, thank you!

hannydevelop commented:
> Hey @hannydevelop , the deploy preview for this PR is failing to show.
> 
> ![image](https://user-images.githubusercontent.com/28389950/96109977-eceb2600-0ee7-11eb-82c1-201bdebf93f2.png)
> 
> I had the same issue just yesterday so @Holzhaus asked me to pull the latest changes from the main repo with `git pull https://github.com/mixxxdj/manual.git manual-2.2.x` so that review will be much simpler. Pull from the main repo then push to your forked repo. That should do the trick :)

Thank you @deborahtrez, the preview works now.

hannydevelop commented:
> alright then, I'll check that tomorrow. Thanks!

Hello @ronso0 all changes have been effected, please review

deborahtrez commented:
> > Hey @hannydevelop , the deploy preview for this PR is failing to show.
> > ![image](https://user-images.githubusercontent.com/28389950/96109977-eceb2600-0ee7-11eb-82c1-201bdebf93f2.png)
> > I had the same issue just yesterday so @Holzhaus asked me to pull the latest changes from the main repo with `git pull https://github.com/mixxxdj/manual.git manual-2.2.x` so that review will be much simpler. Pull from the main repo then push to your forked repo. That should do the trick :)
> 
> Thank you @deborahtrez, the preview works now.

Awesome! Happy to help ;)

Holzhaus commented on source/chapters/effects.rst:
```diff
@@ -27,6 +27,12 @@ processed one after the other in series. You will hear different sounds
 depending on the order of effects within the chain. Each effect in the chain
 has its own button to toggle it on and off.
 
+<<<<<<< HEAD
+The shortcut to quickly Activate or Disable any Effect Unit is :kbd:`5` for Deck  1 and :kbd:`0` in Deck 2.
+=======
+The shortcut to quickly Activate or Disable any Effect Unit is 5 Deck  1 and 0 in Deck 2.
+>>>>>>> b1eb71d86fc5670f886799a713ad47df7f7a4a08
```
This is a [merge conflict marker](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging#_basic_merge_conflicts) and should not have been committed. Please fix.

Holzhaus commented on source/glossary.rst:
```diff
@@ -8,6 +8,12 @@ Glossary of Terms
      Headphone cueing, or just cueing, is listening to the next track you would
      like to mix in your headphones. The audience will not hear what you are
      cueing in your headphones. Being able to cue is a crucial aspect to DJing.
+     The shortcut for headphone cueing in Deck1 is T while for Deck 2 the shortcut is Y.
```
```suggestion
```

I don't think this is needed in the glossary.

Holzhaus commented on source/chapters/library.rst:
```diff
@@ -313,6 +313,13 @@ Tracks can be loaded into a deck in several ways:
+=======
+* Using keyboard shortcut keys to load track in Deck1 with Shift + Left and Deck2
+ with Shift + Right
+>>>>>>> b1eb71d86fc5670f886799a713ad47df7f7a4a08
```
Merge conflict marker.

Holzhaus commented on source/chapters/library.rst:
```diff
@@ -324,6 +331,34 @@ Tracks can be loaded into a deck in several ways:
   Mixxx's waveform displays will load it into that deck.
 
+
+<<<<<<< HEAD
+Unloading Tracks
+================
+
+* To unload track in Deck 1, use the keyboard shortcut :kbd:`Shift` + :kbd:`Left`. In Deck 2 you can
+  unload track with :kbd:`Shift` + :kbd:`Right`
+
+* You can also use the eject track button to unload track from a deck. 
+=======
+UnLoading Tracks
+================
+
+* To unload track in Deck 1, use the keyboard shortcut Shift + Left. In Deck 2 you can
+  unload track with Shift + Right
+
+* You can also use the eject track button to unload track from a deck. Look for a button with this icon:
+
+  .. figure:: ../_static/Mixxx-eject-button.png
+     :align: center
+     :width: 100%
+     :figwidth: 100%
+     :alt: Mixxx library - Eject Button
+     :figclass: pretty-figures
+
+     Mixxx library - Eject Button
+>>>>>>> b1eb71d86fc5670f886799a713ad47df7f7a4a08
```
Merge conflict marker

Holzhaus commented on source/chapters/library.rst:
```diff
@@ -114,7 +114,7 @@ Tracks - View and edit your whole collection
 **Loading tracks**
   To load a track into a :ref:`deck <interface-decks>`, you can either drag it
-  to the waveform display or use the context menu. Go to the chapter
+  to the waveform display, use the context menu, or use shortcut keys. Go to the chapter
```
```suggestion
  to the waveform display, use the context menu or use :term:`keyboard shortcuts <shortcut>`. Go to the chapter
```

Holzhaus commented on source/chapters/user_interface.rst:
```diff
@@ -930,6 +930,10 @@ the slider from left to right. If both decks are playing and the crossfader is
 in its default center position, then you will hear both decks. Right-clicking on
 the crossfader will reset the slider to its default position.
 
+To crossfade left or right in master, you can use the keyboard shortcut G and H 
+in an English Keyboard layout. To apply small crossfade left and small crossfade 
+right in master the shortcut keys are shift+G and shift+H respectively. 
```
```suggestion
To move the crossfader left or right, you can use the :term:`keyboard shortcut` :kbd:`G` and :kbd:`H`.
To apply small crossfade left and small crossfade right, the :term:`shortcuts <shortcut>` are :kbd:`Shift` + :kbd:`G` and :kbd:`Shift` + :kbd:`H` respectively. 
```

Holzhaus commented on source/glossary.rst:
```diff
@@ -8,6 +8,12 @@ Glossary of Terms
+
+   shortcuts
+     Shortcuts let you work more efficiently. This manual shows some shortcuts for 
+     accessing Mixxx functions with an English keyboard layout. Depending on your language
+     settings the defaults may be different.
```
@ronso0 Are the keyboard shortcuts in the translated versions of the manual modified appropriately? If so, I don't know if this is needed.

ronso0 replied:
> Are the keyboard shortcuts in the translated versions of the manual modified appropriately?

I doubt that. Just checked the czech manual (which is hardly translated anyway) and there is no re-mapping of keys.
Also we shouldn't rely on translators updating the keys, so I think the hint here is okay.

hannydevelop replied:
Done! and i used the singular form ```shortcut``` it fits best

hannydevelop commented:
Hello @Holzhaus all changes you requested for has been effected, please review!

ronso0 commented:
@hannydevelop I left some comments on the current state, please go through https://github.com/mixxxdj/manual/pull/135/files and make sure they're all addressed. Thanks!

hannydevelop commented:
> @hannydevelop I left some comments on the current state, please go through https://github.com/mixxxdj/manual/pull/135/files and make sure they're all addressed. Thanks!

All has been addressed, please review @Holzhaus and @ronso0

ronso0 commented on source/chapters/library.rst:
```diff
@@ -324,6 +326,9 @@ Tracks can be loaded into a deck in several ways:
   example, on macOS, dragging-and-dropping a track from iTunes onto one of
   Mixxx's waveform displays will load it into that deck.
 
+
+Unloading Tracks
+
```
Your previous additions are missing
```suggestion

Unloading Tracks
* To unload tracks use the keyboard shortcuts :kbd:`Shift` + :kbd:`Alt` + :kbd:`Left` for Deck 1 and :kbd:`Shift` + :kbd:`Alt` + :kbd:`Right` for Deck 2.

* You can also use the eject track button to unload track from a deck. Look for a button with this icon:

  .. figure:: ../_static/Mixxx-eject-button.png
     :align: center
     :width: 100%
     :figwidth: 100%
     :alt: Mixxx library - Eject Button
     :figclass: pretty-figures
```

hannydevelop replied:
You asked me to remove the picture

ronso0 replied:
sorry, I'm confused some reviews.
I trimmed the suggestion above. please check.

We can add the picture later on, so this get's merged soonish.

Also I went through previous reviews so we don't miss anything and don't have contradicting change proposal.
Sorry for the hazzle!

ronso0 commented on source/chapters/library.rst:
```diff
@@ -347,7 +352,7 @@ a crate, or even the whole library) for tracks that match your search query.
   list of results.
 
-.. note:: If the search input field has focus, the Mixxx keyboard shortcuts are
+.. note:: If the search input field has focus, the Mixxx keyboard :term:`shortcuts <shortcuts>` are
```
This now feels redundant: I think the shortcuts reference is obsolete here when we have "see :ref:`control-keyboard`", so please remove the change.

I'm sorry this gets confusing with multiple reviewers...

hannydevelop replied:
Done!

ronso0 commented:
@hannydevelop I'm sorry this got confusing with multiple reviewers...

Regarding the `shortcuts` link and appendix paragraph @Be-ing suggested:
Unfortunately linking to this is a dead end, and it's confusing when the `shortcuts` link and [using](https://deploy-preview-135--mixxx-manual.netlify.app/chapters/controlling_mixxx.html#using-a-keyboard) are next to each other.
Instead, please link only to [using](https://deploy-preview-135--mixxx-manual.netlify.app/chapters/controlling_mixxx.html#using-a-keyboard) every where this makes sense. And revert the `shortcut` appendix addition.

For a comfortable workflow I recommend you
* adopt the code suggestions from my review
* then locally pull the changes with
```
git fetch [your own remote name] 
git reset --hard [your own remote name]/manual-2.2.x
```
Note that this would wipe any unpushed local changes you did in the meantime.

hannydevelop commented on source/chapters/effects.rst:
```diff
@@ -27,6 +27,8 @@ processed one after the other in series. You will hear different sounds
 depending on the order of effects within the chain. Each effect in the chain
 has its own button to toggle it on and off.
 
+The :term:`shortcut <shortcuts>` to quickly Activate or Disable any Effect Unit is :kbd:`5` for Deck 1 and :kbd:`0` in Deck 2.
```
Sure! will do that now!

hannydevelop replied:
Done!

hannydevelop commented on source/chapters/introduction.rst:
```diff
@@ -27,6 +27,8 @@ is appropriate for the present moment and that will mix well into the
 separate audio outputs, typically provided by a USB audio interface (also known
 as a “sound card”, although most of them are not shaped like cards anymore).
+
+The :term:`shortcut <shortcuts>` for headphone cueing in Deck1 is :kbd:`T` while for Deck 2 the shortcut is :kbd:`Y`. 
```
ok!

hannydevelop replied:
Done!

hannydevelop commented on source/chapters/library.rst:
```diff
@@ -313,6 +313,8 @@ Tracks can be loaded into a deck in several ways:
 * By :ref:`control-keyboard` to load the selected track in the library track
   table.
+* Using keyboard :term:`shortcut <shortcuts>` keys to load track in Deck 1 with :kbd:`Shift` + :kbd:`Left` and Deck 2
+ with :kbd:`Shift` + :kbd:`Right`
```
Ok!

hannydevelop replied:
I don't know if i did this right, please review @ronso0

hannydevelop commented:
Hello @ronso0 all changes have been effected, please review!

ronso0 commented on source/chapters/library.rst:
```diff
@@ -352,7 +349,7 @@ a crate, or even the whole library) for tracks that match your search query.
   list of results.
 
-.. note:: If the search input field has focus, the Mixxx keyboard :term:`shortcuts <shortcuts>` are
+.. note:: note:: If the search input field has focus, the Mixxx keyboard shortcuts are
```
please remove redundant `note::` here

hannydevelop replied:
ooh! Done! Please review @ronso0

ronso0 reviewed (changes requested):
Looks good so far, just one minor change required.

ronso0 commented:
LGTM!
@holzhaus wanna check once more?

ronso0 commented:
@Holzhaus 
Squash and merge is appropriate here considering all this back & forth ?

Holzhaus commented:
Yes.

ronso0 commented:
Great! Thanks @hannydevelop !
